HW8/task2.js

```diff
@@ -0,0 +1,240 @@
+// чтобы ваши функции выполняли строго одну работу. ЭТО ОЧЕНЬ ВАЖНО!
+
+const enterprises = [
+	{
+		id: 1,
+		name: 'Предприятие 1',
+		departments: [
+			{
+				id: 2,
+				name: 'Отдел тестирования',
+				employees_count: 9,
+			},
+			{
+				id: 3,
+				name: 'Отдел маркетинга',
+				employees_count: 20,
+			},
+			{
+				id: 4,
+				name: 'Администрация',
+				employees_count: 15,
+			},
+		],
+	},
+	{
+		id: 5,
+		name: 'Предприятие 2',
+		departments: [
+			{
+				id: 6,
+				name: 'Отдел разработки',
+				employees_count: 50,
+			},
+			{
+				id: 7,
+				name: 'Отдел маркетинга',
+				employees_count: 20,
+			},
+			{
+				id: 8,
+				name: 'Отдел охраны труда',
+				employees_count: 5,
+			},
+		],
+	},
+	{
+		id: 9,
+		name: 'Предприятие 3',
+		departments: [
+			{
+				id: 10,
+				name: 'Отдел аналитики',
+				employees_count: 0,
+			},
+		],
+	},
+]
+
+
+/* Задания:
+	1. Вывести все предприятия и их отделы. Рядом указать количество сотрудников. Для предприятия посчитать сумму всех сотрудников во всех отделах.
+
+	**Пример:**
+
+	Предприятие 1 (45 сотрудников)
+	- Отдел тестирования (10 сотрудников)
+	- Отдел маркетинга (20 сотрудников)
+	- Администрация (15 человек)
+	Предприятие 2 (75 сотрудников)
+	- Отдел разработки (50 сотрудников)
+	- Отдел маркетинга (20 сотрудников)
+	- Отдел охраны труда (5 сотрудников)
+	Предприятие 3 (нет сотрудников)
+	- Отдел аналитики (нет сотрудников)
+*/
+
+function sumOfEmployers(departments) {
+	return departments.reduce((sum, el) => sum + el.employees_count, 0)
+}
+
+function printDepartment (departments) {
+	return departments.map(el => {
+		return `- ${el.name} (${el.employees_count} сотрудников),\n`})
+}
+
+function printEnterprise (enterpise) {
+	return enterpise.map(el => {
+		console.log(`- ${el.name} (${sumOfEmployers(el.departments)} сотрудников): \n ${printDepartment(el.departments)}`)})
+}
+
+printEnterprise(enterprises);
+
+
+// 2. Написать функцию, которая будет принимать 1 аргумент (id отдела или название отдела и возвращать название предприятия, к которому относится).
+// Пример:
+// getEnterpriseName(4) // Предприятие 1
+// getEnterpriseName("Отдел маркетинга") // Предприятие 2
+
+function getEnterpriseName(arg) {
+	const cloneEnterprises = new Object(enterprises).find(ent => ent.departments.find(dep => dep.id === arg || dep.name === arg))
+		
+	return cloneEnterprises ? cloneEnterprises.name : 'Department is not exist'
+}
+
+console.log(getEnterpriseName(10));
+console.log(getEnterpriseName('Отдел маркетинга'))
+
+
+// 3. Написать функцию, которая будет добавлять предприятие. В качестве аргумента принимает название предприятия
+// Пример:
+// addEnterprise("Название нового предприятия")
+
+function findMaxID() {
+	const arr = []
+	for (let i = 0; i < enterprises.length; i++) {
+		arr.push(enterprises[i].id)
+		arr.push(...enterprises[i].departments.map(el => el.id))
+	}
+	return Math.max(...arr)
+}
+
+function addEnterprise(name) {
+	let nextID = findMaxID() + 1;
```

Comment:
better generate new id in this function, e.g. findNewId() where you find the max one and add 1